frontend/apps/reader/modules/readerzooming.lua

```diff
@@ -276,6 +277,23 @@ function ReaderZooming:getZoom(pageno)
     elseif self.zoom_mode == "free" then
         zoom = self.zoom
     end
+    if zoom and zoom > 10 and not Cache:willAccept(zoom * (self.dimen.w * self.dimen.h + 64)) then
+        logger.dbg("zoom too large, adjusting")
+        while not Cache:willAccept(zoom * (self.dimen.w * self.dimen.h + 64)) do
```

Comment:
@poire-z There should be a second condition in here to prevent the possibility of endless loops, but I'm not sure what exactly. A simple `max_iterations = 50`?

Reply:
Dunno. When going towards zero, it should fit isn't it?
But at one point you'll reach zero, or a negative number, so you should probably take care of that too.

Reply:
Good point, a simple > 0 should do the trick.